presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -990,6 +990,26 @@ public void testInvalidAtTimeZone()
         assertFails(TYPE_MISMATCH, "SELECT 'abc' AT TIME ZONE 'America/Los_Angeles'");
     }
 
+    @Test
+    public void testValidJoinOnClause()
+            throws Exception
+    {
+        analyze("SELECT * FROM (VALUES(2,2)) A(X,Y) JOIN (VALUES(2,2)) B(X,Y) ON (TRUE)");
```

Comment:
For readability, add a spaces between VALUES and the list and use lower case for anything that's not a SQL keyword.

Also, the parenthesis around the ON clause are not necessary.